internal/fs/fs_test.go

```diff
@@ -229,6 +231,112 @@ func TestRenameWithFallback(t *testing.T) {
 	}
 }
 
+func TestIsCaseSensitiveFilesystem(t *testing.T) {
```

Comment:
This is a bit inaccurate. You can mount a case-sensitive filesystem on Linux, a case-insensitive one on Linux/macOS...

Can we think of a better way to check this rather than relying on the OS?

Reply:
I agree. But I cannot think of a better way. To some extent, I think we can make some assumptions about the test environment.

Reply:
This will not work on Windows Subsystem for Linux. ```dF -Th``` will give you the actual file system type is the OS is linux.  I'm not sure how it works with samba shares, etc.

Reply:
@colek42 Is Windows Subsystem really something that the tests should handle? Isn't it an edge case we can safely ignore?

Reply:
@sudo-suhas it is easy enough to test for.  See https://github.com/Microsoft/BashOnWindows/issues/423, can you just skip the test if it is WSL?

Reply:
![image](https://user-images.githubusercontent.com/6634325/30172454-dcf439f6-93c2-11e7-8c1c-7fc0575faab7.png)

Reply:
![image](https://user-images.githubusercontent.com/6634325/30172542-21543cae-93c3-11e7-9a45-56fa0ad48e19.png)

Reply:
> Some of us are forced to use Windows.

Ha ha.. Don't worry I am on windows too. I will look into this. Thank you.

Reply:
@sudo-suhas let me know if you need me to test any changes.

Reply:
I'm not sure what's the best way to test this tbh. It's not easy to find a cross-platform solution for this as far as I know...

Reply:
@sdboyer Do I need to check for BashOnWindows? How would I check this in golang? Is it as simple as a file read from `/proc/version`?

Reply:
@sudo-suhas not here - this is just a test. unless we have an actual reason to believe the real logic (not the test logic) fails on WSL, then it's fine to omit here.